pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -170,19 +171,31 @@ func (r *ReconcileMasterUserRecord) ensureUserAccount(logger logr.Logger, murAcc
 			// does not exist - should create
 			userAccount = newUserAccount(nsdName, murAccount.Spec, mur.Spec)
 			if err := memberCluster.Client.Create(context.TODO(), userAccount); err != nil {
-				return false, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToCreateUserAccountReason), err,
+				return 0, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToCreateUserAccountReason), err,
 					"failed to create UserAccount in the member cluster '%s'", murAccount.TargetCluster)
 			}
-			return false, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
+			counter.IncrementUserAccountCount(murAccount.TargetCluster)
+			return 0, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, ""))
 		}
 		// another/unexpected error occurred while trying to fetch the user account on the member cluster
-		return false, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToGetUserAccountReason), err,
+		return 0, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToGetUserAccountReason), err,
 			"failed to get userAccount '%s' from cluster '%s'", mur.Name, murAccount.TargetCluster)
 	}
 	// if the UserAccount is being deleted (by accident?), then we should wait until is has been totally deleted, and this controller will recreate it again
 	if util.IsBeingDeleted(userAccount) {
 		logger.Info("UserAccount is being deleted. Waiting until deletion is complete", "member_cluster", memberCluster.Name)
-		return true, updateStatusConditions(logger, r.client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "recovering deleted UserAccount"))
+		deletionTimestamp := userAccount.GetDeletionTimestamp()
+
+		requeueTime := 3 * time.Second
+		timeUntilDeletion := time.Until(deletionTimestamp.Time)
+		if timeUntilDeletion+3*time.Second >= 0 {
+			counter.DecrementUserAccountCount(logger, murAccount.TargetCluster)
```

Comment:
`counter.DecrementUserAccountCount(logger, murAccount.TargetCluster)` is also called on line 315, is it normal?

Reply:
Yes, it is - this covers the case when the UserAccount is deleted manually, the second one is for the standard deletion.
BTW, I think that you wrote this code :-)

Reply:
yes, I remember about that. That's because we needed to restore a `MUR` in case of manual deletion (cc: @alexeykazakov 😛)